tests/utils/test_hints.py

```diff
@@ -0,0 +1,36 @@
+from typing import Generic, TypeVar
+
+import pytest
+
+from xdsl.utils.hints import get_type_var_mapping
+
+T = TypeVar("T")
+
+
+class GenThing(Generic[T]):
+    pass
+
+
+class GenThingy(GenThing[T]):
+    pass
+
+
+class IntThing(GenThing[int]):
+    pass
+
+
+class IntThingy(GenThingy[int]):
+    pass
```

Comment:
Can you please add more descriptive names to these classes, so it's clear what the purpose is of each part of the generics hierarchy? I'll have a go at naming them to show my current understanding, please correct me if I'm wrong:

```suggestion
class GenericBase(Generic[T]):
    pass


class GenericSubclassOfGenericBase(GenericBase[T]):
    pass


class SpecificSubclassOfGenericBase(GenericBase[int]):
    pass


class SpecificSubclassOfGenericSubclass(GenericSubclassOfGenericBase[int]):
    pass
```